soc/arm/st_stm32/stm32h7/soc.h

```diff
@@ -7,15 +7,15 @@
 #ifndef _STM32F7_SOC_H_
 #define _STM32F7_SOC_H_
 
+#include <sys/util.h>
+
 #ifndef _ASMLANGUAGE
 
+#include <autoconf.h>
 #include <stm32h7xx.h>
 
-/* ARM CMSIS definitions must be included before kernel_includes.h.
- * Therefore, it is essential to include kernel_includes.h after including
- * core SOC-specific headers.
- */
-#include <kernel_includes.h>
+/* Add include for DTS generated information */
+#include <devicetree.h>
```

Comment:
@ioannisg @erwango adjusted this header (hopefully to fix CMSIS problems). I see that some series have the old structure while some others have this one... which is *the correct one*?

Reply:
the correct one is the one we have here now. The problem appears only when MPU is enabled, which makes the build include the cmsis.h headers. That's why some SoCs without MPU can still build properly, but we should change that.